modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/forms/DDMFormHandler.js

```diff
@@ -68,6 +70,37 @@ class DDMFormHandler {
 			}
 		});
 	}
+
+	checkCPInstanceOptions() {
+		var emptyRequiredField = false;
+		const addToCartButton = document.querySelector('.btn-add-to-cart');
```

Comment:
Hi @dsitu , the addToCartButton is a state-driven component. 
Unfortunately, in this case, changing the attributes from the outside is not a valid approach since the changes will be lost in case of a rerendering.

`checkCPInstance` updates the addToCart component via the `${settings.namespace}${CP_INSTANCE_CHANGED}` event. 

I would suggest passing `disabled` in the [payload event](https://github.com/liferay/liferay-portal/blob/0cd3c5abc42a371dc4d587745803a4fd824d0304/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/forms/DDMFormHandler.js#L51)

The function that handles it is [here](https://github.com/liferay/liferay-portal/blob/74241d3d73fb1f34581c450b8d06758a9b1bac0e/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/AddToCart.js#L90)